ui/jstests/listing/test_listing.jsx

```diff
@@ -971,5 +971,146 @@ define(['QUnit', 'jquery', 'react', 'test_utils', 'utils', 'listing'],
       );
 
     });
+
+    QUnit.test('Open and close learning resource panel', function(assert) {
+      var done = assert.async();
+      var learningResourceResponse = {
+        "id": 1,
+        "learning_resource_type": "course",
+        "static_assets": [],
+        "title": "title",
+        "materialized_path": "/course",
+        "content_xml": "<course />",
+        "url_path": "",
+        "parent": null,
+        "copyright": "",
+        "xa_nr_views": 0,
+        "xa_nr_attempts": 0,
+        "xa_avg_grade": 0.0,
+        "xa_histogram_grade": 0.0,
+        "terms": ["required"]
+      };
+      var termResponseEasy = {
+        "id": 1,
+        "slug": "easy",
+        "label": "easy",
+        "weight": 1
+      };
+      var termResponseHard = {
+        "id": 2,
+        "slug": "hard",
+        "label": "hard",
+        "weight": 1
+      };
+      var preReqTermsResponseReq = {
+        "id": 3,
+        "slug": "required",
+        "label": "required",
+        "weight": 1
+      };
+      var preReqTermsResponseNotReq = {
+        "id": 4,
+        "slug": "notrequired",
+        "label": "notrequired",
+        "weight": 1
+      };
+      var vocabularyResponsePrereq = {
+        "id": 1,
+        "slug": "prerequisite",
+        "name": "prerequisite",
+        "description": "Prerequisite",
+        "vocabulary_type": "m",
+        "required": false,
+        "weight": 2147483647,
+        "terms": [preReqTermsResponseReq, preReqTermsResponseNotReq],
+        "multi_terms": true
+      };
+      var vocabularyResponseDifficulty = {
+        "id": 2,
+        "slug": "difficulty",
+        "name": "difficulty",
+        "description": "Difficulty",
+        "vocabulary_type": "f",
+        "required": false,
+        "weight": 2147483647,
+        "terms": [termResponseEasy, termResponseHard],
+        "multi_terms": true
+      };
+      var vocabulariesResponseFirst = {
+        "count": 1,
+        "next": "/api/v1/repositories/test/" +
+        "vocabularies/?type_name=course&page=2",
+        "previous": null,
+        "results": [vocabularyResponseDifficulty]
+      };
+      var vocabulariesResponseSecond = {
+        "count": 1,
+        "next": null,
+        "previous": "/api/v1/repositories/test/" +
+        "vocabularies/?type_name=course",
+        "results": [vocabularyResponsePrereq]
+      };
+
+      TestUtils.initMockjax({
+        url: '/api/v1/repositories/test/learning_resources/' +
+        '1/?remove_content_xml=true',
+        type: 'GET',
+        responseText: learningResourceResponse
+      });
+      TestUtils.initMockjax({
+        url: '/api/v1/repositories/test/vocabularies/?type_name=course',
+        type: 'GET',
+        responseText: vocabulariesResponseFirst
+      });
+      TestUtils.initMockjax({
```

Comment:
Why not use the vocabulary data and mocks already defined at the top of the file?